pkg/vmcp/server/server.go

```diff
@@ -465,3 +568,32 @@ func (s *Server) createPromptHandler(promptName string) func(
 		return result, nil
 	}
 }
+
+// handleHealth handles /health and /ping HTTP requests.
+// Returns 200 OK if the server is running and able to respond.
+//
+// Security Note: This endpoint is unauthenticated and intentionally minimal.
+// It only confirms the HTTP server is responding. No version information,
+// session counts, or operational metrics are exposed to prevent information
+// disclosure in multi-tenant scenarios.
+//
+// For operational monitoring, implement an authenticated /metrics endpoint
+// that requires proper authorization.
+func (*Server) handleHealth(w http.ResponseWriter, _ *http.Request) {
+	response := map[string]string{
+		"status": "ok",
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusOK)
+
+	if err := json.NewEncoder(w).Encode(response); err != nil {
```

Comment:
mm i just realized... we get a failure here, but we are still returning status ok?